src/uu/touch/src/touch.rs

```diff
@@ -88,35 +92,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     ) {
         (Some(reference), Some(date)) => {
             let (atime, mtime) = stat(Path::new(reference), !matches.get_flag(options::NO_DEREF))?;
-            if let Ok(offset) = parse_datetime::from_str(date) {
-                let seconds = offset.num_seconds();
-                let nanos = offset.num_nanoseconds().unwrap_or(0) % 1_000_000_000;
-
-                let ref_atime_secs = atime.unix_seconds();
-                let ref_atime_nanos = atime.nanoseconds();
-                let atime = FileTime::from_unix_time(
-                    ref_atime_secs + seconds,
-                    ref_atime_nanos + nanos as u32,
-                );
-
-                let ref_mtime_secs = mtime.unix_seconds();
-                let ref_mtime_nanos = mtime.nanoseconds();
-                let mtime = FileTime::from_unix_time(
-                    ref_mtime_secs + seconds,
-                    ref_mtime_nanos + nanos as u32,
-                );
-
-                (atime, mtime)
-            } else {
-                let timestamp = parse_date(date)?;
-                (timestamp, timestamp)
-            }
+            let atime = filetime_to_datetime(&atime)
+                .ok_or_else(|| USimpleError::new(1, "failed to convert atime"))?;
```

Comment:
This is too technical as error message. 
Something like "Could not process the reference access time"

Reply:
and it is possible to trigger this error from the touch command ?

Reply:
I've been unable to trigger the error in my testing.

If the duration is too long you get:
```
thread 'main' panicked at 'Duration::seconds out of bounds'
```
in contrast GNU touch reports
```
touch: invalid date format ‘+1000000000000 years’
```
This should probably a new test.

If the new date would be out of bounds, it just doesn't update it, GNU touch seems to behave in the same way.

Reply:
new test is fine :)
thanks

Reply:
I've added the test, unfortunately addressing it is a bit complicated. The error is caused in the dependent library chrono which panics rather than returning a result.